crypto/evp/evp_enc.c

```diff
@@ -336,7 +354,20 @@ int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
             return 0;
         }
     }
+
+    /*
+     * Aliasing in and out in this way is counter intuitive and only supported
+     * for legacy applications.
+     * Don't rely on this to stay the same in the next major version.
+     */
     i = ctx->buf_len;
+    if ((PTRDIFF_T)in != (PTRDIFF_T)(out + i)
+            && is_any_overlapping(out, inl + (bl - 1), in, inl)) {
```

Comment:
One thing where I am not sure is, if the buffer size of the output should be
assumed to be `inl + (bl - 1)` or if it should be `(inl + ctx->buf_len)/bl * bl`
thus if the caller can be expected to know the internal state of the incomplete
buffer.

Reply:
Yeah, I'm pondering over this too...

Reply:
At least the documentation says clearly `inl + (bl - 1)`